### DISCUSSION TITLE
Editing vector shapes

### DISCUSSION THREAD

**Discussion by budaloops** [keiwando/toonsquid#578] [Q&A]
I am practicing calligraphy and soon I want to animate some custom lettering for lyric videos. I am looking for a vector animation app to do this. Between ToonSquid and Linearity Move, I prefer Toonsquid because of its audio feature.

Is there any way to cut a hole into a vector shape that I created in ToonSquid? I have a cursive letter L and I want to cut away part of the shape to create a loop.

![IMG_8522](https://github.com/keiwando/toonsquid/assets/7763208/d8d586ae-23a0-4ccc-9ee5-c8cb5b87ed0c)

I couldn’t find any vector ‘add’ or ‘subtract’ operations in ToonSquid like Affinity Designer has, so is there any workaround/hack to achieve this with the current version of Toonsquid?

SVG import would be great at some point!

**Comment by Cat-Butler** [comment]
Yes, vector functionality in ToonSquid currently somewhat limited, although still very useful for some animation, especially involving camera zooming and significant scaling. So, no masking or vector Booleans yet.

If you rasterise your loop shape(s), then fill a pixel layer or area (encompassing your text) using the loop shapes as a reference (or fill against all layers, with the vector text hidden and the loops layer showing), you will end up with a filled area/layer except for the loops, hide the loops layer, and then use the "everything but the loops" filled layer as a mask for your vector text, to hide the loops.

Unfortunately the mask will be a pixel mask, not a vector mask, but that's the only work around that currently occurs to me, apart from constructing your letter in overlapping pieces that do not include the loop(s).

I'm assuming just colouring your loop areas the same as the background won't work for what you have in mind? (Alternatively, green-screening your text against a background in a separate video editor/compositing app, rather than trying to do it all in ToonSquid)

Possibly somebody else will suggest a better solution that has not occurred to me.

**Reply by budaloops** [reply]
Man these options are so useful. I have been figuring out how to storyboard my animations and I haven’t spent much time thinking of ToonSquid. I’m quite grateful for your help. You sound reasonably experienced. What do you enjoy using ToonSquid / animation apps for?

**Reply by Cat-Butler** [reply]
Currently working on a few minor bits and pieces (as animated GIFs, or static images) for internal training web pages for work.

Personally, I'm very fond of Hanna-Barbera limited animation and stylisation.
Personal project in initial stages (scripting and storyboarding) - comic/cartoon which will be released as both static comic strips and as short animations.  I'm finding using the comic strip idea works surprisingly well to help plan and as a general cinematography guide -  I guess it's really just a variation of the benefits of storyboarding in general.

**Reply by budaloops** [reply]
Sounds like a great use of the comic strip. I hope the cinematography comes thru in the early sketches

**Reply by FoxeyeStudios** [reply]
While there aren’t any ways to make the hole directly on the path object but there is a way to do what you’re describing. First, make the objects as you intended. Then you need to place the larger object layer on top of the smaller object and select toggle mask. Then click on the smaller object’s options and select “hide mask”. Select the options again and select “invert mask”. Then there is a transparent hole in your object.

**Reply by budaloops** [reply]
That’s a neat trick! Much appreciated 🤙

**Reply by Cat-Butler** [reply]
I did not know that "invert mask" appeared as an option - thank you @FoxeyeStudios! 
Definitely a better solution, and it keeps everything as Paths (vector).
